### ISSUE TITLE
Ran 0 batches with an average size of -nan

### ISSUE THREAD

**Issue by delock** [tensorflow/minigo#786]
I met this when I run evaluation with small parallel number of games:
`Ran 0 batches with an average size of -nan `

Is it possible that a ModelBatcher might be destroyed before it run at least one batch?  So far its only a harmless print out that does not cause program termination, but the message is little bit annoying.

the command I use to reproduce this error is like this:
`$ bazel-bin/cc/eval --flagfile=/home/gma/delock-minigo/ml_perf/flags/9.mi/eval.flags --model=tf,/home/gma/delock-minigo/train_log/from_andrew/2019-03-14-22/models/000001-000001.pb --model_two=tf,/home/gma/delock-minigo/train_log/from_andrew/2019-03-14-22/models/target.pb --sgf_dir=/home/gma/delock-minigo/train_log/from_andrew/2019-03-14-22/sgf/eval/target --seed=1`

eval.flags:
```
# eval.flags: Flags for playing eval games.

--flagfile=ml_perf/flags/9.mi/selfplay.flags

# Play fewer games for eval than selfplay.
--parallel_games=2
```

selfplay.flags:
```
# selfplay.flags: Flags for selfplay.

# This flagfile also serves as the base for the boostrap & eval stages of
# the RL loop.

--num_readouts=320
--value_init_penalty=0.2
--holdout_pct=0.03
--disable_resign_pct=0.1
--resign_threshold=-0.99

# Device-specific selfplay flags.
--parallel_games=4
--virtual_losses=8

```

**Comment by tommadams** [comment]
I can't reproduce this behaviour on GPU. I'll rebuild CPU-only TensorFlow now to take a look.

**Comment by tommadams** [comment]
Can you please provide the full output of your command.

**Comment by tommadams** [comment]
Oh. We print `Ran 0 batches with an average size of -nan` near the start of the run, not the end. That was not at all clear from your report.

This is an annoying side effect of how we calculate a model's name (we instantiate a temporary `DualNet` client and call its `name` method). Prior to introducing `RandomDualNet`, we always just assumed that a model's name could be derived from its file path but this is no longer the case.

I'll have a think about whether we can do this a better way.

**Comment by amj** [comment]
@tommadams did #787 fix this?

**Comment by tommadams** [comment]
Nope. I need to refactor some code so that the `Game` object knows the name of the models without having to create dummy instances. It's not as simple as just taking the basename of the model path because the `Random` model doesn't use filenames :/
